core/src/main/java/com/linecorp/armeria/server/logging/LoggingService.java

```diff
@@ -154,23 +163,28 @@ public LoggingService(Service<I, O> delegate, LogLevel level) {
     @Override
     public O serve(ServiceRequestContext ctx, I req) throws Exception {
         if (sampler.isSampled(ctx)) {
-            ctx.log().addListener(log -> logRequest(((ServiceRequestContext) log.context()).logger(),
+            ctx.log().addListener(log -> logRequest(getActiveLogger((ServiceRequestContext) log.context()),
```

Comment:
Could get the active logger outside of the lambda expression, so that it can be reused for the second listener as well.

Reply:
fixed